mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -383,13 +432,55 @@ class AffineParallelLowering : public OpRewritePattern<AffineParallelOp> {
     for (Attribute step : op.steps())
       steps.push_back(rewriter.create<ConstantIndexOp>(
           loc, step.cast<IntegerAttr>().getInt()));
-    // Creating empty scf.parallel op body with appropriate bounds.
-    auto parallelOp = rewriter.create<scf::ParallelOp>(loc, lowerBoundTuple,
-                                                       upperBoundTuple, steps);
-    rewriter.eraseBlock(parallelOp.getBody());
-    rewriter.inlineRegionBefore(op.region(), parallelOp.region(),
-                                parallelOp.region().end());
-    rewriter.eraseOp(op);
+    // Get the terminator operands.
+    Operation *affineParOpTerm = op.region().front().getTerminator();
+    SmallVector<Value, 4> terminatorOperands(
```

Comment:
You don't need to materialize terminatorOperands into a `SmallVector`. You can just use it as a `ValueRange`. Use `getOperand(i)` below and you can do `getNumOperands()` instead of a `size()` on this.